lib_static/plugins/acts_as_journalized/lib/journal_changes.rb

```diff
@@ -41,6 +41,7 @@ def get_changes
 
     @changes.merge!(get_association_changes(predecessor, 'attachable', 'attachments', :attachment_id, :filename))
     @changes.merge!(get_association_changes(predecessor, 'customizable', 'custom_fields', :custom_field_id, :value))
+    @changes.merge!(get_association_changes(predecessor, 'storable', 'file_links', :file_link_id, :link_name))
```

Comment:
This will now be called for all journalized resources which includes work_packages but is not limited to it (e.g. Meeting and WikiPage). The same is of course potentially true for the calls that existed before fetching the changes  to attachments and custom values. E.g. WikiPage does not have a custom values. 

With more associations being checked now, I think it makes sense to introduce simple safeguards to each call to check if the change needs to be collected for this journable or if that would be futile anyway. 

In it's simplest form it could be checking for an included module or method to be there ala `self.class.included_modules.include?(Redmine::Acts::Customizable::InstanceMethods)` but it could also be build into the modules themselves so that we could check for something like `self.class.customized?` (analogous to how it is possible for `journaled?`

Reply:
Agree that it isn't great from a performance perspective, even if the query returns nothing as it is still a FK lookup (which is fast but not so fast as not running).

That makes me think of a couple of changes:

1. `Journal::CreateService` could probably tailor its SQL based on `journable` capabilities (i.e. not run the extra sql for attachables etc).
2. `JournalChanges`, through journal, could ask the `journable` and only run what's necessary.
3. We could implement the extra methods on `Journal` itself, basically asking the `journable` if it responds to `attachments`, `custom_fields` or `file_links`. It isn't perfect, but has low attrition (as for now they would be only relevant to the journal).

What do you think?

Reply:
I see that you now started on 2. to improve the code in `JournalChanges` and that is by far enough for this PR. Doing 1. makes a ton of sense since we should be conscious of performance with this service but it needn't be in this PR. For motivation, though: When the dates of a  work package being part of an elaborate ancestry/follows relationship is updated, the other work packages being part of the relationship might also have to be updated. I have seen cases where because of a date change to a work package 50+ other work packages were updated. In such a case, every ms of improvement counts.